crates/common/src/signer/schemes/bls.rs

```diff
@@ -0,0 +1,105 @@
+use alloy::rpc::types::beacon::{
+    constants::BLS_DST_SIG, BlsPublicKey as BlsPublicKeyInner, BlsSignature,
+};
+use blst::BLST_ERROR;
+use derive_more::derive::{Deref, Display, From, Into, LowerHex};
+use serde::{Deserialize, Serialize};
+use ssz_derive::{Decode, Encode};
+use tree_hash::TreeHash;
+use tree_hash_derive::TreeHash;
+
+use crate::{
+    error::BlstErrorWrapper, signature::sign_builder_root, signer::GenericPubkey, types::Chain,
+    utils::blst_pubkey_to_alloy,
+};
+
+pub type BlsSecretKey = blst::min_pk::SecretKey;
+
+// std traits
+#[derive(Debug, Clone, Copy, LowerHex, Display, PartialEq, Eq, Hash)]
+// serde, ssz, tree_hash
+#[derive(Serialize, Deserialize, Encode, Decode, TreeHash)]
+#[ssz(struct_behaviour = "transparent")]
+// derive_more
+#[derive(Deref, From, Into)]
+pub struct BlsPublicKey {
+    inner: BlsPublicKeyInner,
```

Comment:
I had a quick thought about what we can do regarding opening an issue upstream, but I decided for a wrapper type as a common solution. Now that you propose it, I will have a deeper look into how easy this would be (or maybe if the trait is intentionally not implemented); and eventually open up an issue.

Reply:
Another possible argument in favour of a wrapper type would be SDK unification, because we currently have a custom type for the `EcdsaPublicKey` (since `k256::ecdsa` doesn't export a convenient type to represent an encoded point as we need it; like alloy does for `BlsPublicKey`). 
So, currently, module authors would include the `EcdsaPublicKey` from our common crate, but they'll have to import the proper version of `BlsPublicKey` directly from alloy (and care about matching versions). So I'd say this makes sense for us to reexport as a custom wrapper type in that regard, although it does, indeed, add a bit of complexity. 🤔

Reply:
Opened up the issue [here](https://github.com/sigp/tree_hash/issues/22). 

Implementing it locally wasn't a problem, but then we'll have some versioning issues after it gets merged, which tie into [this task](https://github.com/Commit-Boost/commit-boost-client/issues/36).

Leaving the wrapper type as a temporary solution for now.